app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -1218,4 +1223,47 @@ private void setFullscreenModeSdkLessThan30() {
                         | View.SYSTEM_UI_FLAG_FULLSCREEN
         );
     }
+
+    /**
+     * Helper class for the barcode zoom scalers.
+     */
+    public class SeekBarListener implements SeekBar.OnSeekBarChangeListener {
+        private SeekBar seekBar;
+
+        public SeekBarListener(SeekBar sb) {
+            seekBar = sb;
+        }
+
+        public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+            if (!fromUser) {
+                Log.d(TAG, "non user triggered onProgressChanged, ignoring, progress is " + progress);
+                return;
+            }
+            Log.d(TAG, "Progress is " + progress);
+            Log.d(TAG, "Max is " + seekBar.getMax());
+            float scale = (float) progress / (float) seekBar.getMax();
+            Log.d(TAG, "Scaling to " + scale);
+
+            if (seekBar == barcodeHeightScaler) {
+                loyaltyCard.zoomLevel = progress;
+                DBHelper.updateLoyaltyCardZoomLevel(database, loyaltyCardId, loyaltyCard.zoomLevel);
+                setCenterGuideline(loyaltyCard.zoomLevel);
+            } else if (seekBar == barcodeWidthScaler) {
+                loyaltyCard.zoomWidth = progress;
+                DBHelper.updateLoyaltyCardZoomWidth(database, loyaltyCardId, loyaltyCard.zoomWidth);
+                mainImage.getLayoutParams().width = mainLayout.getWidth() * loyaltyCard.zoomWidth / 100;
+                mainImage.requestLayout();
+            }
```

Comment:
It seems better to me if this class gets told with a separate parameter if it's working on height and width so it doesn't break in unexpected ways if and variable name changes and so it doesn't depend on variable names outside of itself.